packages/core/src/html/SiteLinkManager.ts

```diff
@@ -69,4 +74,57 @@ export class SiteLinkManager {
     this._addToCollection(resourcePath, cwf);
     return 'Intralink collected to be validated later';
   }
+
+  /**
+   * Add sections that could be reached by intra-link with hash to this node to FilePathToHashesMap,
+   * The reachable sections include nodes with ids and headings.
+   *
+   * ForceWrite should only be called when processing heading node with the maintainHashesForInclude method.
+   */
+  maintainFilePathToHashesMap(node: MbNode, cwf: string, forceWrite: string = '') {
+    if (!this.config.intrasiteLinkValidation.enabled) {
+      return;
+    }
+    const path = cwf.substring(this.config.rootPath.length);
+    if (!this.filePathToHashesMap.has(path)) {
+      this.filePathToHashesMap.set(path, new Set());
+    }
+    if (forceWrite !== '') {
+      this.filePathToHashesMap.get(path)!.add(forceWrite);
+    } else if (node.attribs!.id) {
+      this.filePathToHashesMap.get(path)!.add(node.attribs!.id);
+    }
+  }
+
+  printFilePathToHashesMap() {
```

Comment:
Reminder to rename this based on @EltonGohJH's comment!

Reply:
Method deleted.